api/app/signals/apps/email_integrations/settings.py

```diff
@@ -0,0 +1,53 @@
+from django.conf import settings
+from django.test.signals import setting_changed
+
+DEFAULTS = {
+    'INTEGRATIONS'
+}
+
+
+class AppSettings(object):
+    def __init__(self, user_settings=None, defaults=None):
+        if user_settings:
+            self._user_settings = user_settings
+        self.defaults = defaults or DEFAULTS
+        self._cached_attrs = set()
+
+    @property
+    def user_settings(self):
+        if not hasattr(self, '_user_settings'):
+            self._user_settings = getattr(settings, 'MAIL_INTEGRATION_SETTINGS', {})
+        return self._user_settings
+
+    def __getattr__(self, attr):
+        if attr not in self.defaults:
+            raise AttributeError('Invalid mail integration setting: {}'.format(attr))
+
+        try:
+            val = self.user_settings[attr]
+        except KeyError:
+            val = self.defaults[attr]
+
+        # Cache the result
+        self._cached_attrs.add(attr)
+        setattr(self, attr, val)
```

Comment:
Not a fan of setting properties on the AppSetting object itself (may lead to overwriting parts of the object). Is the cache ever accessed because of the overridden `__getattr__` method?

Reply:
The `__getattr__` is only called if an attribute does not exists on the object. See https://docs.python.org/3/reference/datamodel.html#object.__getattr__